Doc/library/functions.rst

```diff
@@ -458,9 +458,9 @@ are always available.  They are listed here in alphabetical order.
 
 .. function:: eval(expression[, globals[, locals]])
 
-   The arguments are a string and optional globals and locals.  If provided,
-   *globals* must be a dictionary.  If provided, *locals* can be any mapping
-   object.
+   The arguments are a string or a compiled code object, optionally along
+   with globals and locals.  If provided, globals must be a dictionary and
+   locals can be any mapping object.
```

Comment:
The next paragraph still seems to assume that expression is a string.

Reply:
@iritkatriel What makes you think that?

Reply:
Code objects are not parsed.

Reply:
> The next paragraph still seems to assume that expression is a string.

@iritkatriel What do you think about correcting the first sentence in the next paragraph in this way?

_The expression argument is if not compiled code object, it parsed and evaluated as a
Python expression (technically speaking, a condition list) using the globals and
locals dictionaries as global and local namespace._

Reply:
That doesn't seem grammatically correct. Typos?

Reply:
_If the expression argument is not a compiled code object, it is parsed and evaluated as a Python expression (technically speaking, a list of conditions) using the globals and locals dictionaries as the global and local namespace._

@iritkatriel I made a correction in the sentence. Can you review it again?

Reply:
I'd say "if the expression is a string" rather than if it's not a code object. But the second part happens to code objects too. So you need to sentences: (1) if it's a string it is compiled to a code object. (2) Then the code object is evaluated.

Reply:
_If the expression is a string, it is compiled to a code object. Then the code object is parsed and evaluated as a Python expression (technically speaking, a list of conditions) using the globals and locals dictionaries as the global and local namespace._

@iritkatriel I made a correction in the sentence. Can you review it again?